### PR TITLE
[GPU] Fix a bug to miscalculate feature-axis index for fused-ops

### PR DESCRIPTION
**Pull request by jade-cho**
### Tickets:
 - *96780*

### CODE REVIEW

**Review by isanghao — CHANGES_REQUESTED** [review]
Please add a test case

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 1]
```diff
@@ -1603,6 +1603,35 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, conv_fp32_activation_eltwise_diff_sizes, :
...
+    execute(p);
+}
+
+// in_shape; out_shape; eltw_shape; kernel; stride; pad; dilation; groups; data_type; input_format; weights_type; weights_format; default_type; default_format;
+#define CASE_GROUP_CONV_ELTW_FP32_1 { 1, 48, 3, 3 }, { 1, 48, 3, 3 }, { 1, 48, 1, 1 }, { 1, 1, 3, 3 }, { 1, 1 }, { 1, 1 }, { 1, 1 }, 16, data_types::f32, format::bfyx, data_types::f32, format::g_os_iyx_osv16, data_types::f32, format::bfyx
+#define CASE_GROUP_CONV_ELTW_FP32_2 { 1, 48, 3, 3 }, { 1, 48, 3, 3 }, { 1, 48, 3, 3 }, { 1, 1, 3, 3 }, { 1, 1 }, { 1, 1 }, { 1, 1 }, 16, data_types::f32, format::bfyx, data_types::f32, format::g_os_iyx_osv16, data_types::f32, format::bfyx
+
+INSTANTIATE_TEST_SUITE_P(fusings_gpu, conv_fp32_group_conv_eltwise_sum, ::testing::ValuesIn(std::vector<conv_eltw_test_params>{
+    conv_eltw_test_params{ CASE_GROUP_CONV_ELTW_FP32_1, 2, 2, 3 },
+    conv_eltw_test_params{ CASE_GROUP_CONV_ELTW_FP32_2, 3, 2, 3 },
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:1632]: This test passes even without your fix. Please confirm that it fails without your patch.

**Review by isanghao — CHANGES_REQUESTED** [review]
Please see my comment.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 1 above)
Comment (jade-cho) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: Updated. This test will be failed with --device_suffix=1 without my patch.

**Review by isanghao — APPROVED** [review]

**Pull request merged**